ios-apps-master/Beer_Diary/Beer_Diary/Add_Beer.swift

```diff
@@ -0,0 +1,256 @@
+        newItem.aroma = tempVal
+        aroma.text = "Aroma - \(tempVal)/10"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderAppearance(_ sender: Any)
+    {
+        let tempVal = Int(appearanceSlider.value)
+        appearanceSlider.value = Float(tempVal)
+        newItem.appearance = tempVal
+        appearance.text = "Appearance - \(tempVal)/5"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderPalate(_ sender: Any)
+    {
+        let tempVal = Int(palateSlider.value)
+        palateSlider.value = Float(tempVal)
+        newItem.palate = tempVal
+        palate.text = "Palate - \(tempVal)/5"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderTaste(_ sender: Any)
+    {
+        let tempVal = Int(tasteSlider.value)
+        tasteSlider.value = Float(tempVal)
+        newItem.taste = tempVal
+        taste.text = "Taste - \(tempVal)/10"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    
+    @IBAction func add(_ sender: Any)
+    {
+        if let string = beerName.text
+        {
+            newItem.name = string
+        }
+        if let string = breweryName.text
+        {
+            newItem.brewery = string
+        }
+        if let string = styleName.text
+        {
+            newItem.style = string
+        }
+        if let string = countryName.text
+        {
+            newItem.country = string
+        }
+        if let string = thoughts.text
+        {
+            newItem.thoughts = string
+        }
+        
+        
+        
+       if (newItem.aroma != 0) && (newItem.appearance != 0) && (newItem.palate != 0) && (newItem.taste != 0) && (newItem.name != "") && (newItem.brewery != "") && (newItem.style != "") && (newItem.country != "") && (newItem.thoughts != "")
+       {
+            newItem.date = NSDate()
+        
+            BeerList.append(newItem)
+        
+            //UserDefaults.standard.set(BeerList, forKey: "BeerList")
+        
+            let alert = UIAlertController(title: "Beer Added", message: "", preferredStyle: UIAlertControllerStyle.alert)
+        
+            alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: { (action) in
+                self.dismiss(animated: true, completion: nil)
+            }))
+        
+            saveData()
+        
+            self.present(alert, animated: true, completion: nil)
```

Comment:
It's a bad UX design, to make the user tap `OK` in an alert with only one option. It's better to use alerts when there is an important choice. This can be notified to the user using for example `MBProgressHUD`